algorithms/src/std_algorithms/impl/Kokkos_UniqueCopy.hpp

```diff
@@ -78,38 +77,105 @@ OutputIterator unique_copy_impl(const std::string& label,
-OutputIterator unique_copy_impl(const std::string& label,
-                                const ExecutionSpace& ex, InputIterator first,
-                                InputIterator last, OutputIterator d_first) {
+OutputIterator unique_copy_exespace_impl(const std::string& label,
+                                         const ExecutionSpace& ex,
+                                         InputIterator first,
+                                         InputIterator last,
+                                         OutputIterator d_first) {
   // checks
   Impl::static_assert_random_access_and_accessible(ex, first, d_first);
   Impl::static_assert_iterators_have_matching_difference_type(first, d_first);
   Impl::expect_valid_range(first, last);
 
+  // aliases
+  using value_type1   = typename InputIterator::value_type;
+  using value_type2   = typename OutputIterator::value_type;
+  using binary_pred_t = StdAlgoEqualBinaryPredicate<value_type1, value_type2>;
+
+  // run
+  return unique_copy_exespace_impl(label, ex, first, last, d_first,
+                                   binary_pred_t());
+}
+
+//
+// team level
+//
+
+template <class TeamHandleType, class InputIterator, class OutputIterator,
+          class PredicateType>
+KOKKOS_FUNCTION OutputIterator unique_copy_team_impl(
+    const TeamHandleType& teamHandle, InputIterator first, InputIterator last,
+    OutputIterator d_first, PredicateType pred) {
+  // checks
+  Impl::static_assert_random_access_and_accessible(teamHandle, first, d_first);
+  Impl::static_assert_iterators_have_matching_difference_type(first, d_first);
+  Impl::expect_valid_range(first, last);
+
+  // branch for trivial vs non trivial case
+  const std::size_t num_elements = Kokkos::Experimental::distance(first, last);
+  if (num_elements == 0) {
+    return d_first;
+  } else if (num_elements == 1) {
+    d_first[0] = first[0];
+    return d_first + 1;
+  } else {
+    // FIXME: parallel_scan is what we used for the execution space impl,
+    // but parallel_scan does not support TeamThreadRange, so for the
+    // team-level impl we do this serially for now and later figure out
+    // if this can be done in parallel
+
+    std::size_t count = 0;
+    Kokkos::View<std::size_t, typename TeamHandleType::execution_space>
+        countView(&count);
+    Kokkos::single(Kokkos::PerTeam(teamHandle), [=]() {
+      for (std::size_t i = 0; i < num_elements - 1; ++i) {
+        const auto& val_i   = first[i];
+        const auto& val_ip1 = first[i + 1];
+        if (!pred(val_i, val_ip1)) {
+          d_first[countView()++] = val_i;
+        }
+      }
+
+      // we need to copy the last element always
+      d_first[countView()++] = first[num_elements - 1];
+    });
+    teamHandle.team_barrier();
+
+    // return the correct iterator: we need +1 here because we need to
+    // return iterator to the element past the last element copied
+    return d_first + count;
```

Comment:
this is definitely wrong. Only thread 0 will get back a count != 0. You need to use a broadcast single. Here, we also should improve the test figuring out why this doesn't fail the test.